server/sonar-server/src/test/java/org/sonar/server/authentication/JwtSerializerTest.java

```diff
@@ -74,12 +74,27 @@ public void generate_token_with_expiration_date() throws Exception {
     underTest.start();
     Date now = new Date();
 
-    String token = underTest.encode(new JwtSession(USER_LOGIN, 10));
+    long expirationTimeInSeconds = 10L;
+    String token = underTest.encode(new JwtSession(USER_LOGIN, expirationTimeInSeconds));
+
+    assertThat(token).isNotEmpty();
+    Claims claims = underTest.decode(token).get();
+    assertThat(claims.getExpiration().getTime()).isGreaterThanOrEqualTo(now.getTime() + expirationTimeInSeconds * 1000L - 1000L);
+  }
+
+  @Test
+  public void generate_token_with_big_expiration_date() throws Exception {
+    setSecretKey(A_SECRET_KEY);
+    underTest.start();
+    Date now = new Date();
+
+    long oneYearInSeconds = 12 * 30 * 24 * 60 * 60L;
```

Comment:
As discussed, this class doesn't check the value of the session, so the limit should be LONG#MAX_VALUE